app/routines/newflow/create_or_update_salesforce_lead.rb

```diff
@@ -39,36 +39,71 @@ def exec(user:)
         adoption_json = build_book_adoption_json_for_salesforce(user)
       end
 
-      lead = OpenStax::Salesforce::Remote::Lead.new(
-        first_name: user.first_name,
-        last_name: user.last_name,
-        phone: user.phone_number,
-        email: user.best_email_address_for_salesforce,
-        source: LEAD_SOURCE,
-        application_source: DEFAULT_REFERRING_APP_NAME,
-        role: sf_role,
-        position: sf_position,
-        title: user.other_role_name,
-        who_chooses_books: user.who_chooses_books,
-        subject_interest: user.which_books,
-        num_students: user.how_many_students,
-        adoption_status: ADOPTION_STATUS_FROM_USER[user.using_openstax_how],
-        adoption_json: adoption_json,
-        os_accounts_id: user.id,
-        accounts_uuid: user.uuid,
-        school: user.most_accurate_school_name,
-        city: user.most_accurate_school_city,
-        country: user.most_accurate_school_country,
-        verification_status: user.faculty_status == User::NO_FACULTY_INFO ? nil : user.faculty_status,
-        b_r_i_marketing: user.is_b_r_i_user?,
-        title_1_school: user.title_1_school?,
-        newsletter: user.receive_newsletter?,
-        newsletter_opt_in: user.receive_newsletter?,
-        sheerid_school_name: user.sheerid_reported_school,
-        instant_verification: user.is_sheerid_verified,
-        account_id: sf_school_id,
-        school_id: sf_school_id
-      )
+      user.faculty_status = User::INCOMPLETE_SIGNUP unless user.is_profile_complete?
+
+      lead = OpenStax::Salesforce::Remote::Lead.find_by(email: user.best_email_address_for_salesforce)
```

Comment:
You could probably do
```
lead = OpenStax::Salesforce::Remote::Lead.find_by(email: user.best_email_address_for_salesforce) || OpenStax::Salesforce::Remote::Lead.new(email: user.best_email_address_for_salesforce)
```

and then you don't need the `if` as the logic is the same after that

Reply:
Tried this, but unfortunately, it needs the school to create a new one, so it doesn't work with the `||`.
Maybe I can add the school into that call... but let me think about the implications of that (like if the school gets updated).

Reply:
@Dantemss I think it's best (not a fan of the duplication) to keep the `if` statement. I changed it slightly to check if the user has a `salesforce_lead_id` if so, search for them by email, which I think will cover some of the lead merging that could happen on the salesforce side.
Let me know if you think otherwise -- ready for another look when you have a chance.